### PR TITLE
[FEAT] Events Expand Button

### PR DESCRIPTION
**Pull request by hv7214**
**Details**
1. Added expand-collapse button to reduce the scroll time.
2. Removed paging mechanism which resulted in all events to occur under the same metric. 
3. Added `Events count` column, which represents the number of events associated to the metric.
4. Since now all events are rendered on a single page, it would be very uncomfortable UX when there are lots of events. So added `scroll-to-top` button for events list.

**UI changes**
```
Total metrics count: 106
Total events count: 22000+
```
Peek 2020-03-26 15-234

**Note**

There are **no such performance issues**, though it may look a bit laggy that's because of low GIF quality(10 FPS).

**Related issue**

https://github.com/moira-alert/moira/issues/363

### CODE REVIEW

**Comment by hv7214** [comment]
@sashasushko  @beevee can you please review this : )

**Comment by beevee** [comment]
You're quite brave to tackle an issue with Russian task description :-)
I've translated it for you just now, see moira-alert/moira#363.

I'm not sure this PR solves the problem, though. Original issue description states that user is confused seeing the same metric multiple times on different pages. To solve this issue, you should either get rid of paging (and collapse events so they don't make page overly long), or generate groups on backend.

Try testing this on a trigger with large number of metrics and events (say, 100s of metrics with 100s of events each). Make sure that everything works fast and doesn't hang your browser.

**Comment by hv7214** [comment]
@beevee I guess I understood it right : ) so to solve this problem I compressed events into one showing the last event, with a dropdown button to show all events associated with the metric.

Will make sure whether it works fine with 100s of metrics with 100s of events, though I think React will handle this without any hassle : )

**Comment by beevee** [comment]
> to solve this problem I compressed events into one showing the last event

Yes, but you still request events 100 at a time. So you will have multiple pages. There is nothing to prevent a single metric from appearing multiple times on different pages. Your code does nothing to address this.

**Comment by hv7214** [comment]
@beevee I've a doubt, why metric will appear multiple times ? (sorry if this question is silly!) 

I just thought, if events count go very high then it will cause an issue for the user to scroll down deep to find the metric. So made that expand-collapse button which reduces the same overhead.

Thanks!

**Comment by beevee** [comment]
@hv7214 try adding some test data to your installation. Add two metrics and several events for these metrics with interleaving timestamps:

```
metric1 event1
metric2 event1
metric1 event2
metric2 event2
...
```

If I understand this issue correctly, as soon as you add enough events for two pages (say, 150 events), you will see that both metric1 and metric2 appear on both pages.

**Comment by hv7214** [comment]
@beevee Thanks for the clearance! I'll surely try this : )

**Comment by hv7214** [comment]
@beevee I tried,  you were right. I have to find another solution to this : )

**Comment by hv7214** [comment]
@beevee i was thinking to merge both solutions as given in https://github.com/moira-alert/moira/issues/363. in fact, I've done second solution which in turn solves the issue of events covering the whole page. What's say : ) ?

**Comment by beevee** [comment]
@hv7214 I think you should just get rid of paging and render all events as one big page

**Comment by hv7214** [comment]
@beevee it seems a good idea, but removing the paging mechanism will resulting in a deep scroll. Don't you think it may degrade the user experience.

**Comment by beevee** [comment]
@hv7214 several comments earlier you've literally told me that:

> React will handle this without any hassle : )

I'm no expert in React, so I don't know. Perhaps, you should write a test, like I suggested several comments earlier:

> Try testing this on a trigger with large number of metrics and events (say, 100s of metrics with 100s of events each). Make sure that everything works fast and doesn't hang your browser.

**Comment by hv7214** [comment]
@beevee okay, will give it a shot then let you know about the results : )

**Comment by hv7214** [comment]
@beevee does it look good : ) ?

**Comment by hv7214** [comment]
@beevee have a look!

**Comment by beevee** [comment]
@hv7214 I assume that you did a test with hundreds of metrics + hundreds of events and got an acceptable rendering speed in your browser?

**Comment by hv7214** [comment]
@beevee just did with 2 metrics + 100s of events. Sorry for that, will tell you the results of above soon

**Comment by beevee** [comment]
@hv7214 please, re-read all comments carefully before pinging me again, so I don't have to repeat the same suggestion three times ([first](https://github.com/moira-alert/web2.0/pull/305#issuecomment-598043928), [second](https://github.com/moira-alert/web2.0/pull/305#issuecomment-599987430), [third](https://github.com/moira-alert/web2.0/pull/305#issuecomment-603760556)).

**Comment by hv7214** [comment]
Btw I have an answer why it will work seamlessly: On an average, the number of components decides the rendering speed, And since the number of components is equal to the number of metrics. So I think it won't affect much. Then too, will soon provide the benchmarks for the above : )

**Comment by hv7214** [comment]
@beevee I've updated benchmarks tests(see in the description). 
There are no such performance issues, collapse button made rendering fast.

Btw do you think it's a good idea to include the event count column because all events are hidden(except the last one)?

**Comment by beevee** [comment]
@hv7214 yes, adding events count is a good idea.

File: src/Api/MoiraApi.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ export default class MoiraApi implements IMoiraApi {
 
     triggerListPageSize: number = 20;
 
-    eventHistoryPageSize: number = 100;
+    eventHistoryPageSize: number = 0;
```
Comment (hv7214) [inline src/Api/MoiraApi.js]: fetches all events

File: src/Components/EventList/EventList.jsx [hunk 2]
```diff
@@ -1,15 +1,12 @@
...
 import type { Event } from "../../Domain/Event";
-import StatusIndicator from "../StatusIndicator/StatusIndicator";
-import roundValue from "../../helpers/roundValue";
 import cn from "./EventList.less";
+import MetricEvents from "../MetricEvents/MetricEvents";
 
 type Props = {|
     items: {
-        [key: string]: Array<Event>,
+        [key: string]: Array<Array<Event>, Number>,
```
Comment (hv7214) [inline src/Components/EventList/EventList.jsx]: Now items will store the event list and event count

**Comment by hv7214** [comment]
@beevee @sashasushko can you please do the final review : )

Thank you.

**Comment by hv7214** [comment]
Closing this PR because commits got messed up. Will make new PR soon : )

**Pull request closed without merging**